composer/loggers/neptune_logger.py

```diff
@@ -0,0 +1,231 @@
+
+        from neptune.utils import stringify_unsupported
+
+        assert self._base_handler is not None
+
+        self._base_handler[NeptuneLogger.TRACE_NAMESPACE] = stringify_unsupported(traces)
+
+    def can_upload_files(self) -> bool:
+        """Whether the logger supports uploading files."""
+        return True
+
+    def upload_file(
+        self,
+        state: 'State',
+        remote_file_name: str,
+        file_path: pathlib.Path,
+        *,
+        overwrite: bool = False,
+    ):
+        if not self._enabled or not self._log_artifacts:
+            return
+
+        assert self._neptune_run is not None
+
+        neptune_path = f'{self._base_namespace}/{remote_file_name}'
+        if self._neptune_run.exists(neptune_path) and not overwrite:
+
+            warnings.warn(f"The file '{neptune_path}' already exists and overwrite is set to False."
+                          'No action will be taken.')
+            return
+
+        del state  # unused
+
+        assert self._base_handler is not None
+        self._base_handler[remote_file_name].upload(str(file_path))
+
+    def download_file(
+        self,
+        remote_file_name: str,
+        destination: str,
+        overwrite: bool = False,
+        progress_bar: bool = True,
+    ):
+        del progress_bar  # not supported
+
+        if not self._enabled:
+            return
+
+        if os.path.exists(os.path.join(
+                destination,
+                remote_file_name,
+        )) and not overwrite:
+            warnings.warn(f"Destination '{destination}' already exists and overwrite is set to False."
+                          'No action will be taken.')
+            return
+
+        assert self._neptune_run is not None
+        assert self._base_handler is not None
+
+        file_path = f'{self._base_namespace}/{remote_file_name}'
+        if not self._neptune_run.exists(file_path):
+            raise FileNotFoundError(f'File {file_path} not found')
+
+        self._base_handler[remote_file_name].download(destination=destination)
+
+    def log_images(
+        self,
+        images: Union[np.ndarray, torch.Tensor, Sequence[Union[np.ndarray, torch.Tensor]]],
+        name: str = 'Images',
+        channels_last: bool = False,
+        step: Optional[int] = None,
+        masks: Optional[Dict[str, Union[np.ndarray, torch.Tensor, Sequence[Union[np.ndarray, torch.Tensor]]]]] = None,
+        mask_class_labels: Optional[Dict[int, str]] = None,
+        use_table: bool = True,
+    ):
+        if not self._enabled:
+            return
+
+        assert self._base_handler is not None
+
+        if not isinstance(images, Sequence) and images.ndim <= 3:
+            images = [images]
+
+        for img in images:
+            self._base_handler[name].append(img, step=step)
+
+    def post_close(self) -> None:
+        if not self._enabled:
+            return
+
+        assert self._neptune_run is not None
+        self._neptune_run.sync()
```

Comment:
Do we explicitly need to call `sync()` before `stop()`?

Reply:
Afaik `stop()` doesn't sync the data - on the contrary, I think it actually kills a synchronization thread